src/intrinsics/ecma262/ArrayPrototype.js

```diff
@@ -825,6 +870,30 @@ export default function(realm: Realm, obj: ObjectValue): void {
     // 1. Let O be ? ToObject(this value).
     let O = To.ToObject(realm, context);
 
+    // if there is an initialValue, then we cannot be sure
+    // that the return value is an array and will like
+    // have side-effects so this is not supported for now
+    if (Widen.hasWidenedNumericUnknownProperty(O)) {
+      if (initialValue || initialValue !== realm.intrinsics.undefined) {
+        let diagnostic = new CompilerDiagnostic(
+          "array reduce with initial value is not supported",
+          realm.currentLocation,
+          "PP0034",
```

Comment:
This error code has already been taken. Update it and create a wiki page for it before someone else takes PP0035.